src/main/scala/mesosphere/marathon/api/akkahttp/v2/PodsController.scala

```diff
@@ -55,26 +63,28 @@ class PodsController(
       (extractClientIP & forceParameter) { (clientIp, force) =>
         extractRequest { req =>
           entity(as[raml.Pod]) { podDef =>
-            val normalizedPodDef = podNormalizer.normalized(podDef)
-            val pod = Raml.fromRaml(normalizedPodDef).copy(version = clock.now())
-            assumeValid(PodsValidation.pluginValidators(pluginManager).apply(pod)) {
-              authorized(CreateRunSpec, pod).apply {
-                val p = async {
-                  val deployment = await(podManager.create(pod, force))
-
-                  // TODO: How should we get the ip?
-                  val ip = clientIp.getAddress().toString
-                  eventBus.publish(PodEvent(ip, req.uri.toString(), PodEvent.Created))
-
-                  deployment
-                }
-                onSuccess(p) { plan =>
-                  val ramlPod = PodConversion.podRamlWriter.write(pod)
-                  val responseHeaders = Seq(
-                    Location(Uri(pod.id.toString)),
-                    Headers.`Marathon-Deployment-Id`(plan.id)
-                  )
-                  complete((StatusCodes.Created, responseHeaders, ramlPod))
+            assumeValid(podDefValidator().apply(podDef)) {
+              val normalizedPodDef = podNormalizer.normalized(podDef)
+              val pod = Raml.fromRaml(normalizedPodDef).copy(version = clock.now())
+              assumeValid(PodsValidation.pluginValidators(pluginManager).apply(pod)) {
+                authorized(CreateRunSpec, pod).apply {
+                  val p = async {
+                    val deployment = await(podManager.create(pod, force))
+
+                    // TODO: How should we get the ip?
+                    val ip = clientIp.getAddress().toString
+                    eventBus.publish(PodEvent(ip, req.uri.toString(), PodEvent.Created))
+
+                    deployment
+                  }
+                  onSuccess(p) { plan =>
```

Comment:
I would go with something better than just "p" what do you think?

Reply:
today's code is brought to you today by the letter P
P is for:
passionate
programmer
precise
pod
programming
party
probably?
Is it Pyscho to Prefer to Pick a Properly Pedantic  Protean?